pkg/encryption/utils.go

```diff
@@ -76,7 +80,15 @@ func SignedValue(seed string, key string, value []byte, now time.Time) string {
 func cookieSignature(signer func() hash.Hash, args ...string) string {
 	h := hmac.New(signer, []byte(args[0]))
 	for _, arg := range args[1:] {
-		h.Write([]byte(arg))
+		_, err := h.Write([]byte(arg))
+		// If signing fails, fail closed and return something that won't validate
```

Comment:
Do we want to attempt to return random data that won't validate as a SHA1 when this does start working (or validate with itself if SHA1 continues to break?)

Or just immediately `logger.Fatal` if we can't Hash and not bother having a fallback?

Reply:
How hard do you think it would be to be able to return an error from this function? That would allow us to do something like returning an internal server error without having to make up some behaviour here

Reply:
Gave it a whirl here: https://github.com/oauth2-proxy/oauth2-proxy/pull/690/commits/e4cec484f8e7f79cba657d79c35e31366674d88d